tests/rest/client/test_third_party_rules.py

```diff
@@ -34,40 +40,44 @@
 
 
 class LegacyThirdPartyRulesTestModule:
-    def __init__(self, config: Dict, module_api: "ModuleApi"):
+    def __init__(self, config: Dict, module_api: "ModuleApi") -> None:
         # keep a record of the "current" rules module, so that the test can patch
         # it if desired.
         thread_local.rules_module = self
         self.module_api = module_api
 
     async def on_create_room(
         self, requester: Requester, config: dict, is_requester_admin: bool
-    ):
+    ) -> bool:
         return True
 
-    async def check_event_allowed(self, event: EventBase, state: StateMap[EventBase]):
+    async def check_event_allowed(
+        self, event: EventBase, state: StateMap[EventBase]
+    ) -> Union[bool, dict]:
         return True
 
     @staticmethod
-    def parse_config(config):
+    def parse_config(config: Dict[str, Any]) -> Dict[str, Any]:
         return config
 
 
 class LegacyDenyNewRooms(LegacyThirdPartyRulesTestModule):
-    def __init__(self, config: Dict, module_api: "ModuleApi"):
+    def __init__(self, config: Dict, module_api: "ModuleApi") -> None:
         super().__init__(config, module_api)
 
-    def on_create_room(
+    def on_create_room(  # type: ignore[override]
```

Comment:
I got always
```
tests/rest/client/test_third_party_rules.py:68: error: Return type "bool" of "on_create_room" incompatible with return type "Coroutine[Any, Any, bool]" in supertype "LegacyThirdPartyRulesTestModule"  [override]
```

I did not seen the `async` and will do an update.

Reply:
Thanks! FWIW: The return type `Coroutine[...]` is a clue that there's something non-synchronous going on (since async functions are regular functions which return a coroutine behind the scenes).